### ISSUE TITLE
helm-projectile-grep with Tramp mode is broken

### ISSUE THREAD

**Issue by GamovCoder** [bbatsov/helm-projectile#76]
## Expected behavior
After `helm-projectile-grep` has returned a list of results, a user can jump to any of those results.

## Actual behavior
The path to a selected result is ill-formed. E.g. If Tramp mode is used over ssh, the path is formed like this: `/ssh:username@remote-address:/ssh:username@remote-address:/home/username/target_result.cpp`. The wrong part here is `/ssh:username@remote-address:/ssh:username@remote-address:`. It is repeated twice.

## Steps to reproduce the problem

1. Run `helm-projectile-grep` while using Tramp mode. 
2. Select a result and press `C-j` to preview the file.
3. Watch a new buffer has been created.

## Environment & version information

- `helm-projectile` version:  20170202.1000
- `helm` version (in `helm-pkg.el`):20170209.318
- `projectile` version (`M-x projectile-version`):  20170106.606
- Emacs version (`M-x emacs-version`): 25.1.1
- OS: Ubuntu 14.04, OS X Sierra

*Note*: `projectile-grep` works fine for me.

**Comment by kgrimes2** [comment]
+1

**Comment by mikej96** [comment]
I can reproduce this one as well. projectile-grep works as expected.

**Comment by mcqueenjordan** [comment]
+1, encountered this and found the issue while googling the behavior.

**Comment by Kypert** [comment]
@xiongtx, seems like it depends on how the results from the [git] grep is returned. The PR i made in helm-grep.el is a workaround, but I suppose another way is to adjust the search results to list the absolute path.

Currently I am thinking that it boils down to the `expand-file-name` function in `helm-projectile-grep-or-ack` (https://github.com/Kypert/helm/blob/f633966ee4640ba27fc0eaa794e67b0bfe2450a1/helm-grep.el#L1161):

```
;; Works, this is what is produced for e.g helm-find-files -> C-s:
;; /workspace/git/me/proto/main.c
(expand-file-name "/workspace/git/me/proto/main.c" "/ssh:such-wow.se:/workspace/git/me/proto/")

;; helm-projectile-grep makes helm-grep.el add an additional tramp prefix 
;; because below is not a "local file"
;; /ssh:such-wow.se:/workspace/git/me/proto/main.c
(expand-file-name "./main.c" "/ssh:such-wow.se:/workspace/git/me/proto/")
```
Since I am quite new to both helm and helm-projectile, I am wondering if you have any comments here, where is the better place to resolve this? :) For my daily use, I can use the workaround, but seems like it might not be accepted as-is? Can I help out more?

**Comment by Kypert** [comment]
Fixed by emacs-helm/helm/commit@4b912fed5533ae4cd98dc6fa26ce1ec8251d8947.

**Comment by xiongtx** [comment]
@Kypert Thanks! Do you think this issue can be closed?
